### PR TITLE
macdeploy: alternative info to download the macOS SDK

### PR DESCRIPTION
**Pull request by darosior**
The previous link wasn't accessible for me, this adds some instructions
given to me by Hebasto on #bitcoin-core-builds as well as a shasum for
the archive to quickly check the downloaded one is the right one before
processing with the entire Guix build.

### CODE REVIEW

File: contrib/macdeploy/README.md [hunk 1]
```diff
@@ -15,8 +15,10 @@ When complete, it will have produced `Bitcoin-Core.dmg`.
 Our current macOS SDK
 (`Xcode-12.1-12A7403-extracted-SDK-with-libcxx-headers.tar.gz`) can be
 extracted from
-[Xcode_12.1.xip](https://download.developer.apple.com/Developer_Tools/Xcode_12.1/Xcode_12.1.xip).
+[Xcode_12.1.xip](https://download.developer.apple.com/Developer_Tools/Xcode_12.1/Xcode_12.1.xip)
+(alternatively, after login to your account go to "Downloads", then "More" and look for [`Xcode_12.1`]( https://download.developer.apple.com/Developer_Tools/Xcode_12.1/Xcode_12.1.xip)).
```
Comment (hebasto) [inline contrib/macdeploy/README.md:19]: extra space?
```suggestion
(alternatively, after login to your account go to "Downloads", then "More" and look for [`Xcode_12.1`](https://download.developer.apple.com/Developer_Tools/Xcode_12.1/Xcode_12.1.xip)).
```

Comment (darosior) [reply contrib/macdeploy/README.md]: Woops, done.

**Comment by hebasto** [comment]
Concept ACK.

It seems https://developer.apple.com just requires that one is logged in before following the https://download.developer.apple.com/Developer_Tools/Xcode_12.1/Xcode_12.1.xip link.

**Comment by Emzy** [comment]
Concept ACK.
I always get a "403 Forbidden", also after being logged in.
Searching for "12.1" on https://developer.apple.com/download/all/ works.
I checked the sha256sum of my Xcode_12.1.xip file and got a match. 612443b1894b39368a596ea1607f30cbb0481ad44d5e29c75edb71a6d2cf050f

**Comment by hebasto** [comment]
While touching this file, maybe update these lines as well https://github.com/bitcoin/bitcoin/blob/5bbb2edf6960798eee6998181fa78f0bbe07740e/contrib/macdeploy/README.md#L79-L81 ?

Or just drop them?

File: contrib/macdeploy/README.md [hunk 2]
```diff
@@ -15,8 +15,10 @@ When complete, it will have produced `Bitcoin-Core.dmg`.
 Our current macOS SDK
 (`Xcode-12.1-12A7403-extracted-SDK-with-libcxx-headers.tar.gz`) can be
 extracted from
-[Xcode_12.1.xip](https://download.developer.apple.com/Developer_Tools/Xcode_12.1/Xcode_12.1.xip).
+[Xcode_12.1.xip](https://download.developer.apple.com/Developer_Tools/Xcode_12.1/Xcode_12.1.xip)
+(alternatively, after login to your account go to "Downloads", then "More" and look for [`Xcode_12.1`](https://download.developer.apple.com/Developer_Tools/Xcode_12.1/Xcode_12.1.xip)).
```
Comment (jonatack) [inline contrib/macdeploy/README.md:19]: Would there be any point in linking to this one? 

https://bitcoincore.org/depends-sources/sdks/Xcode-12.1-12A7403-extracted-SDK-with-libcxx-headers.tar.gz

Comment (jarolrod) [reply contrib/macdeploy/README.md:19]: you're not allowed to redistribute the SDK, that's a fake link 🥸

**Comment by achow101** [comment]
I think it would be useful to mention that cookies for https://download.developer.apple.com/ need to be allowed in order for the actual download link to work. Otherwise you could still login and click the download link and still get a 403.

**Review by jarolrod — COMMENTED** [review]
Concept ACK

It's a bit tricky to accurately convey my suggestions using the online github tools, so below is a diff and [here is a branch](https://github.com/jarolrod/bitcoin/tree/pr22516-alt) with my suggestions:
```diff
diff --git a/contrib/macdeploy/README.md b/contrib/macdeploy/README.md
index ad52f7d20..6b8de576e 100644
--- a/contrib/macdeploy/README.md
+++ b/contrib/macdeploy/README.md
@@ -15,10 +15,11 @@ When complete, it will have produced `Bitcoin-Core.dmg`.
 Our current macOS SDK
 (`Xcode-12.1-12A7403-extracted-SDK-with-libcxx-headers.tar.gz`) can be
 extracted from
-[Xcode_12.1.xip](https://download.developer.apple.com/Developer_Tools/Xcode_12.1/Xcode_12.1.xip)
-(alternatively, after login to your account go to "Downloads", then "More" and look for [`Xcode_12.1`](https://download.developer.apple.com/Developer_Tools/Xcode_12.1/Xcode_12.1.xip)).
-An Apple ID is needed to download this.
-The `sha256sum` of the archive should be `612443b1894b39368a596ea1607f30cbb0481ad44d5e29c75edb71a6d2cf050f`.
+[Xcode_12.1.xip](https://download.developer.apple.com/Developer_Tools/Xcode_12.1/Xcode_12.1.xip). An Apple ID is needed to download this.
+
+Alternatively, after logging in to your Apple Developer account, go to 'Downloads' -> 'More' and look for [`Xcode_12.1`](https://download.developer.apple.com/Developer_Tools/Xcode_12.1/Xcode_12.1.xip).
+
+The `sha256sum` of the SDK archive file should be `612443b1894b39368a596ea1607f30cbb0481ad44d5e29c75edb71a6d2cf050f`.
 
 After Xcode version 7.x, Apple started shipping the `Xcode.app` in a `.xip`
 archive. This makes the SDK less-trivial to extract on non-macOS machines. One

```

**Comment by darosior** [comment]
> While touching this file, maybe update these lines as well 
> Or just drop them?

As mentioned afterward by Achow the link isn't dead, but needs cookies to be enabled. Edited to reflect that.

> I think it would be useful to mention that cookies for https://download.developer.apple.com/ need to be allowed in order for the actual download link to work. Otherwise you could still login and click the download link and still get a 403.

Done.

> so below is a diff and here is a branch with my suggestions

Thanks, took part of it for the english and formatting part.

Diff:
```patch
diff --git a/contrib/macdeploy/README.md b/contrib/macdeploy/README.md
index ad52f7d20e..cdf2b6609f 100644
--- a/contrib/macdeploy/README.md
+++ b/contrib/macdeploy/README.md
@@ -15,8 +15,10 @@ When complete, it will have produced `Bitcoin-Core.dmg`.
 Our current macOS SDK
 (`Xcode-12.1-12A7403-extracted-SDK-with-libcxx-headers.tar.gz`) can be
 extracted from
-[Xcode_12.1.xip](https://download.developer.apple.com/Developer_Tools/Xcode_12.1/Xcode_12.1.xip)
-(alternatively, after login to your account go to "Downloads", then "More" and look for [`Xcode_12.1`](https://download.developer.apple.com/Developer_Tools/Xcode_12.1/Xcode_12.1.xip)).
+[Xcode_12.1.xip](https://download.developer.apple.com/Developer_Tools/Xcode_12.1/Xcode_12.1.xip).
+Cookies must be enabled for this link to work after logging in.
+Alternatively, after logging in to your account go to 'Downloads', then 'More'
+and look for [`Xcode_12.1`](https://download.developer.apple.com/Developer_Tools/Xcode_12.1/Xcode_12.1.xip).
 An Apple ID is needed to download this.
 The `sha256sum` of the archive should be `612443b1894b39368a596ea1607f30cbb0481ad44d5e29c75edb71a6d2cf050f`.
 ```

File: contrib/macdeploy/README.md (hunk 2 above)
Comment (luke-jr) [inline contrib/macdeploy/README.md:19]: Might be good to also note cookies need to be enabled for the hostname.

Comment (darosior) [reply contrib/macdeploy/README.md]: Done.
```diff
diff --git a/contrib/macdeploy/README.md b/contrib/macdeploy/README.md
index cdf2b6609f..c2cc7c58b3 100644
--- a/contrib/macdeploy/README.md
+++ b/contrib/macdeploy/README.md
@@ -16,10 +16,9 @@ Our current macOS SDK
 (`Xcode-12.1-12A7403-extracted-SDK-with-libcxx-headers.tar.gz`) can be
 extracted from
 [Xcode_12.1.xip](https://download.developer.apple.com/Developer_Tools/Xcode_12.1/Xcode_12.1.xip).
-Cookies must be enabled for this link to work after logging in.
 Alternatively, after logging in to your account go to 'Downloads', then 'More'
 and look for [`Xcode_12.1`](https://download.developer.apple.com/Developer_Tools/Xcode_12.1/Xcode_12.1.xip).
-An Apple ID is needed to download this.
+An Apple ID and cookies enabled for the hostname are needed to download this.
 The `sha256sum` of the archive should be `612443b1894b39368a596ea1607f30cbb0481ad44d5e29c75edb71a6d2cf050f`.
 
 After Xcode version 7.x, Apple started shipping the `Xcode.app` in a `.xip`
```

**Comment by hebasto** [comment]
> > While touching this file, maybe update these lines as well
> > Or just drop them?
> 
> As mentioned afterward by Achow the link isn't dead, but needs cookies to be enabled. Edited to reflect that.

Another link points at Xcode_11.3.1 but we use Xcode_12.1 for now.

**Review by Imanuelc112 — APPROVED** [review]

**Comment by fanquake** [comment]
Do we really need to link to the same `xip` from 3 different places in this README? Given that half the time the links don't even seem to work (they don't work for me now even when logged in), why not just remove them, or just have one.

As mentioned above, can you also update the instance of `Xcode_11.3.1` further down, and remove the link there as well. There is some duplication of information in this file, but that doesn't have to be fixed here.

**Comment by darosior** [comment]
> Do we really need to link to the same xip from 3 different places in this README?

No, i don't think so. I figured adding another link was helpful since it's what worked for me but i definitely don't have any opinion on how to rework this doc. This PR was just meant as a quick tiny link fix as i got through the process..

**Review by fanquake — APPROVED** [review]
ACK f8f772dc495f7d3ee02d75d42ee22ceb90c3a693

**Pull request merged**